src/components/WeatherWidget/WeatherWidget.tsx

```diff
@@ -62,15 +58,20 @@ export default function WeatherWidget({ location }: Props) {
         setIsLoading(false);
       }
     };
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
     fetchForecast();
-  }, [enqueueSnackbar, location, selectedLanguage, selectedTemperatureUnit, t]);
+  }, [enqueueSnackbar, location, language, temperatureUnit, t]);
+
+  if (isLoading) {
+    return (
+      <div css={classes.spinner}>
+        <Spinner />
+      </div>
+    );
+  }
 
-  return isLoading ? (
-    <CircularProgress size={SPINNER_SIZE} />
-  ) : (
-    !!forecast && (
-      <>
+  if (forecast) {
```

Comment:
Return `null` explicitly if there is no forecast. Also enforce rule `consistent-return` in ESLint